power_grid_model_c/power_grid_model/include/power_grid_model/component/fault.hpp

```diff
@@ -141,15 +157,25 @@ class Fault final : public Base {
     double x_f_;
 
     void check_sanity() const {
-        if (fault_type_ == FaultType::three_phase) {
-            if (fault_phase_ != FaultPhase::abc) {
-                throw InvalidShortCircuitPhases(fault_type_, fault_phase_);
-            }
-        }
-        else {
-            if (fault_phase_ == FaultPhase::abc) {
-                throw InvalidShortCircuitPhases(fault_type_, fault_phase_);
+        auto supported = [](FaultType fault_type) -> std::vector<FaultPhase> {
```

Comment:
Is using a dynamic vector here too much? Can you find a way of only using a stack variable? Maybe an `std::array` with maximum length?

Reply:
i can try but that feels like overengineering to me. i'll have a look

Reply:
At this moment all the component classes do not have any dynamic allocation. It would be good to keep in this way.

Reply:
I feel like using the whole lambda and a `std::find` itself is a bit too much? Maybe just a simple switch on `fault_type` and each case just a simple if throw statement is enough?

Reply:
i tried that but it got incredibly verbose. can you have a quick look at my new solution? i think this one looks quite clean and can be resolved compile-time (aside from the `throw` statements)

Reply:
Your current implementation should be fine.